### PR TITLE
PT-2465: Anonymised pedigree should not show external ID, comments, names and DoB.

### PR DESCRIPTION
**Pull request by veronikaslc**
Fixed.

### CODE REVIEW

**Comment by veronikaslc** [comment]
@allasm @mjshepherd please, review

File: components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js [hunk 1]
```diff
@@ -811,12 +811,17 @@ define([
                     this.getAgeLabel().show();
                     labels.push(this.getAgeLabel());
                 }
+                this.getExternalIDLabel() && labels.push(this.getExternalIDLabel());
```
Comment (allasm) [inline components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js]: Don't we need to .show() and .hide() getExternalIDLabel() as well?

File: components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js [hunk 2]
```diff
@@ -811,12 +811,17 @@ define([
...
+                if (this.getCommentsLabel()) {
+                    this.getCommentsLabel().show();
+                    labels.push(this.getCommentsLabel());
+                }
             } else {
                 this.getNameLabel() && this.getNameLabel().hide();
                 this.getAgeLabel() && this.getAgeLabel().hide();
+                this.getExternalIDLabel() && labels.push(this.getExternalIDLabel());
```
Comment (allasm) [inline components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js]: we don't want to do labels.push in this case

File: components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js (hunk 1 above)
Comment (veronikaslc) [inline components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js]: Hm.. In JIRA issue external ID isn't mentioned. Can it be considered as sensitive info?

File: components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js (hunk 2 above)
Comment (veronikaslc) [inline components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js]: But if we need to show external ID?
This line originaly was implemented regardless anonimization, thats why it is both if and else blocks to preserve an original order of lines in pdf.

Comment (allasm) [inline components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js]: AH, I see, I did not realize what happened. You are right, order is important in this case. But I would probably stop the if-else, do one line for externalID, do another if-else specifically for comments. Code duplication is IMHO worse and less intuitive and less maintainable than an extra if

Comment (veronikaslc) [inline components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js]: Did that already in previous PR, Sergiu did not like this way.

Comment (sdumitriu) [inline components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js]: There's always something to duplicate, either the `if`, or the `push`. I find it clearer if there's only one `if` statement.

An alternative is to move the external ID as the first label, before the `if`.

Comment (veronikaslc) [inline components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js]: Then questions to discuss: 
 1. Do we need to hide external id in anonimized pdf?
 2. Can we change the order of labels to make external id the first?

Comment (veronikaslc) [inline components/pedigree/resources/src/main/resources/pedigree/view/personVisuals.js]: In today's stand-up it was decided to merge this fix as it is, and create an issue for next milestone to refactor the pedigree anonimization to be fetched from configuration.

**Comment by veronikaslc** [comment]
Updated PR commit.

**Comment by marta-** [comment]
The whole anonymization implementation should be improved to have the PII fields all listed in one place (ideally configurable) and have them all treated in a generic way, instead of putting little pieces of code related to specific fields here and there. This is probably not a show stopper for this PR, but [a refactoring](https://phenotips.atlassian.net/browse/PT-2476) would drastically improve the maintainability of this code.

**Pull request merged**